llpc/context/llpcPipelineContext.cpp

```diff
@@ -393,6 +408,16 @@ void PipelineContext::setOptionsInPipeline(Pipeline *pipeline) const {
                     "Mismatch");
       shaderOptions.fp32DenormalMode = static_cast<lgc::DenormalMode>(shaderInfo->options.fp32DenormalMode);
       shaderOptions.adjustDepthImportVrs = shaderInfo->options.adjustDepthImportVrs;
+      // disableLicmThreshold is set to the first of:
+      // - a non-zero value from the corresponding shaderInfo option
+      // - a value of 1 if DisableLicm or the disableLicm shaderInfo option is true
+      // - the value of DisableLicmThreshold
+      // Default is 0, which does not disable LICM
+      shaderOptions.disableLicmThreshold =
+          shaderInfo->options.disableLicmThreshold
+              ?: ((DisableLicm || shaderInfo->options.disableLicm) ? 1 : DisableLicmThreshold);
```

Comment:
Don't use `?:`, it's a non-standard GNU extension. Can you also rewrite this so that there are no nested conditions within one expression?

Reply:
Fixed.